domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -429,13 +434,37 @@ class TopicListController @Inject constructor(
     )
   }
 
-  private fun getDependentLisOfTopicForOngoingTopic(topicId: String): List<String> {
+  // TODO(#2550): Remove hardcoded order of topics. Compute list of suggested stories from backend structures
+  /** Returns a list of topic IDs for which the specified topic ID expects to be completed before being suggested. */
+  private fun retrieveTopicDependencies(topicId: String): List<String> {
     val listOfTopicIds = mutableListOf<String>()
-    if (topicId == TEST_TOPIC_ID_0) {
-      listOfTopicIds.add(FRACTIONS_TOPIC_ID)
-    } else if (topicId == TEST_TOPIC_ID_1) {
-      listOfTopicIds.add(TEST_TOPIC_ID_0)
-      listOfTopicIds.add(RATIOS_TOPIC_ID)
+    when (topicId) {
+      FRACTIONS_TOPIC_ID -> {
+        listOfTopicIds.add(TEST_TOPIC_ID_0)
```

Comment:
This doesn't seem correct. Fractions should depend on addition + subtraction, multiplication, and division (per your document). My earlier comment suggested making topic 0 depend on fractions, and topic 1 depend on topic 0 & ratios (that way the test topics are not part of the production behavior & you can still test the multi-topic scenarios). Please fix.

Reply:
Corrected.